### PR TITLE
Fix rotodrop top piece replication issue (#16791)

### PR DESCRIPTION
**Pull request by qwzhaox**
Fix rotodrop top piece replication issue (#16791). 
- Add a loop that iterates until the TileElement is of type track.

I was able to reproduce the bug and confirmed that the bug was fixed and nothing else in relation to the rotodrop was broken (was still able to rotate / finish construction / build another rotodrop). Let me know if there's anything else I need to do!

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
I gave it a little test and it looks to work and not break the top blocking that the community uses: 
image
THIS PR


image
Develop

**Comment by qwzhaox** [comment]
> Please add a changelog entry and also raise the network version

Done! Hopefully I did it right.

**Comment by duncanspumpkin** [comment]
As this is drawing only the network doesn't need to be bumped but doesn't matter.

File: distribution/changelog.txt [hunk 1]
```diff
@@ -28,6 +28,7 @@
 - Fix: [#19950] Mine train block brake supports drawn incorrectly.
 - Fix: [#19987] [Plugin] ‘SetCheatAction’ has wrong ID in plugin API.
 - Fix: [#19991] Crash using cut-away view.
+- Fix: [#16791] Rotodrop top piece replicates when walls are placed around it and clearance checks are disabled.
```
Comment (duncanspumpkin) [inline distribution/changelog.txt:31]: we keep this in order of the issue so this needs to move up to L17. Should also mention that this is an "(original bug)" see below how that is normally marked.

**Comment by ZehMatt** [comment]
> As this is drawing only the network doesn't need to be bumped but doesn't matter.

I think all clients should see the same so there wont be any confusion, but yeah the simulation is technically not altered, it doesn't hurt anyway.

File: distribution/changelog.txt (hunk 1 above)
Comment (qwzhaox) [reply distribution/changelog.txt]: Moved, thanks for letting me know

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**